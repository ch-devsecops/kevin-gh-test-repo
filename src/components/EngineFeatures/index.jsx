import React from 'react';
import PropTypes from 'prop-types';
import ProductFeatures from '../ProductFeatures';
import getValueByKey from '../../utils/getValueByKey';

const EngineFeatures = ({ fields, rendering }) => {
  if (!fields) return null;

  const gtmTags = {
    type: rendering?.componentName,
  };

  const trimId = getValueByKey(fields, 'model')[0]?.fields?.detIdentifier?.value;
  if (!trimId) return null;

  return <ProductFeatures trimId={trimId} gtmTags={gtmTags} />;
};

EngineFeatures.propTypes = {
  fields: PropTypes.shape({
    detIdentifier: PropTypes.shape({
      value: PropTypes.string,
    }),
  }),
};

export default EngineFeatures;
