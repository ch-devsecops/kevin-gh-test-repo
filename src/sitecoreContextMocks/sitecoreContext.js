const commonSitecoreContext = {
  pageEditing: false,
  pageState: 'normal',
  language: 'en',
  sitecoreApiHost: 'https://app-honda-pod01-cd-plan-dmm-ccentral-dit.azurewebsites.net',
  localhostSitecoreApiHost: 'https://app-honda-pod02-cd-plan-dmm-ccentral-dit.azurewebsites.net',
  sitecoreApiKey: '453A4BB7-B80C-4351-9173-CF0748C3793F',
  hondaRestApiHost: 'uat-api.honda.ca',
  settings: { defaultProvince: 'ON' },
  sotHondaExperienceApiConfig: {
    consumerName: 'HCICONSUMERWEB',
    consumerId: '1000001',
    env: 'sit',
  },
};

const sitecoreContext = {
  engine: {
    site: {
      name: 'Engine',
    },
    ...commonSitecoreContext,
  },
  marine: {
    site: {
      name: 'Marine',
    },
    ...commonSitecoreContext,
  },
  honda: {
    site: {
      name: 'Honda',
    },
    ...commonSitecoreContext,
  },
  acura: {
    site: {
      name: 'Acura',
    },
    ...commonSitecoreContext,
  },
  mc: {
    site: {
      name: 'MC',
    },
    ...commonSitecoreContext,
  },
  psp: {
    site: {
      name: 'PSP',
    },
    partStreamConfig: {
      host: '//services.arinet.com/PartStream/',
      appKey: 'Vvsd42YoyPLGcuiWQlqj',
      id: 'aripartstream',
    },
    ...commonSitecoreContext,
  },
};

export default sitecoreContext;
