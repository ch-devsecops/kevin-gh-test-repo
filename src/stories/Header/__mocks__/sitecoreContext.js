import dictionary from '../../../sitecoreContextMocks/locales/en/translation.json';

export default {
  dictionary,
  route: {
    placeholders: {
      header: [
        {
          uid: 'aa61ffb8-7220-4df4-aa8c-0a6f9abff222',
          componentName: 'SiteLogo',
          dataSource: '{A9BA3BCE-93E8-4DA5-9DD2-552987B8691D}',
          params: {},
          fields: {
            siteLogo: {
              value: {
                src: './sample-mock-data/media/img/placeholders/logo-placeholder.png',
                alt: 'logo',
                width: '115',
                height: '37',
              },
            },
            headerUrl: {
              value: {
                href: 'http://instagram.com/hondamotoca',
                linktype: 'external',
                url: 'http://instagram.com/hondamotoca',
              },
            },
          },
        },
      ],
    },
  },
  language: 'en',
  languageSelectors: [
    {
      code: 'en',
      url: '/en/',
    },
    {
      code: 'fr',
      url: '/fr/',
    },
  ],
};
