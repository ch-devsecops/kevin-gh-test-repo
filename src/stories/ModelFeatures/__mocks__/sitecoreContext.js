import * as dictionary from '../../../sitecoreContextMocks/locales/en/translation.json';

const defaultSitecoreContext = {
  dictionary,
  route: {
    fields: {
      isDarkMode: {
        value: false,
      },
    },
  },
  language: 'en',
  languageSelectors: [
    {
      code: 'en',
      url: '/en/',
    },
    {
      code: 'fr',
      url: '/fr/',
    },
  ],
};

export const darkSitecoreContext = {
  ...defaultSitecoreContext,
  route: { fields: { isDarkMode: { value: true } } },
};

export default defaultSitecoreContext;
