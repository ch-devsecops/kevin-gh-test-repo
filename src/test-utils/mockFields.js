/* eslint-disable max-len */
/* eslint-disable max-classes-per-file */

// FIXME: jsdoc popup not working on test files
/** Class representing a mock data. */
class MockData {
  /**
   * Add a field.
   * @param {string} key - key of the field obj: i.e. 'title'
   * @param {Object} value - value of the field obj: i.e. { value: 'some text' }.
   */
  add(key, value) {
    this[key] = value;
  }
}

/**
 * Class representing mock fields object.
 * @extends MockData
 *
 * available method: add()
 *
 * default fields: [anchorId, title, bodyText, image, ctaLink, ctaType, ctaIcon, gtmCategory, gtmTitle, contentTitle, subtitle]
 *
 * note: we can add more commonly used fields in the class declaration if needed
 */
class MockFields extends MockData {
  constructor() {
    super();

    this.title = { value: 'Title' };
    this.bodyText = { value: 'BodyText' };
    this.image = {
      value: { src: 'mock.png', alt: 'mock alt' },
    };
    this.ctaLink = {
      value: {
        href: 'www.source.ca',
        title: 'cta aria label',
        text: 'CTA Text',
      },
    };
    this.ctaType = {
      value: 'Tertiary',
    };
    this.ctaIcon = {
      fields: {
        value: {
          value: 'arrowRight',
        },
      },
    };
    this.gtmCategory = {
      fields: {
        value: {
          value: 'Others',
        },
      },
    };
    this.gtmTitle = {
      value: 'gtm title',
    };
    this.contentTitle = {
      value: 'Content Title',
    };
    this.subtitle = {
      value: 'Content Subtitle',
    };
    this.seoH1 = {
      value: '',
    };
    this.anchorId = {
      value: 'anchor-id',
    };
    this.ctaLabel = {
      value: 'CTA Label',
    };
  }
}

/**
 * Class representing a mock rendering object
 * @extends MockData
 *
 * available method: add()
 */
class MockRendering extends MockData {
  /**
   * Create a mock rendering object
   * @param {string} componentName - adds componentName
   */
  constructor(componentName) {
    super();

    this.componentName = componentName;
  }
}

export { MockFields, MockRendering };
