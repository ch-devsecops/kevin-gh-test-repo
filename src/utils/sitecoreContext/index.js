export { default as useApiHost } from './useApiHost';
export { default as useApiUrl } from './useApiUrl';
export { default as useAppName } from './useAppName';
export { default as useDarkFromRoute } from './useDarkFromRoute';
export { default as useLanguage } from './useLanguage';
export { default as usePageEditing } from './usePageEditing';
export { default as usePageState } from './usePageState';
export { default as usePartStreamConfig } from './usePartStreamConfig';
export { default as useProvinces } from './useProvinces';
export { default as useProvincesPath } from './useProvincesPath';
export { default as useRoute } from './useRoute';
export { default as useSettings } from './useSettings';
export { default as useSharedApps } from './useSharedApps';
export { default as useSitecoreApiHost } from './useSitecoreApiHost';
export { default as useSitecoreApiKey } from './useSitecoreApiKey';
