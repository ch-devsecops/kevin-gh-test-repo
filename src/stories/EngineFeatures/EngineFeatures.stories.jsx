import React from 'react';
import { Box } from '@honda-canada/design-system-react';
import EngineFeatures from '../../components/EngineFeatures';
import mockData from './componentProps';
import sitecoreContext from '../../sitecoreContextMocks/sitecoreContext';

export default {
  title: 'Engine Features',
  component: EngineFeatures,
};

const sitecoreContextMock = sitecoreContext.engine;

const Template = ({ storyProps }) => (
  <Box mx="l">
    <EngineFeatures {...storyProps} />
  </Box>
);

export const Default = Template.bind({});
Default.args = { storyProps: mockData, context: sitecoreContextMock };
