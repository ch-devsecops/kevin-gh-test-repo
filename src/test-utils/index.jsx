import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { SitecoreContext } from '@sitecore-jss/sitecore-jss-react';
import { BrowserRouter } from 'react-router-dom';
import { OptionalVideoModalContextProvider } from '@honda-canada/design-system-react';
import UserLocationProvider from '@honda-canada/user-location';
import { configureAxe } from 'jest-axe';
import ModelExplorationContextProvider from '../components/ModelExplorationContext';
import ModelFiltersContextProvider from '../components/ModelFiltersContext';
import { MockFields, MockRendering } from './mockFields';
import MockTheme from './mockTheme';
import 'intersection-observer';
import { ACURA_THEME_NAME } from '../utils/constants';

const generateJssWrapper = context =>
  // eslint-disable-next-line react/display-name
  function ({ children }) {
    return (
      <SitecoreContext
        context={{
          language: 'en',
          site: { name: ACURA_THEME_NAME },
          ...context?.sitecoreContext,
        }}
        componentFactory={{}}
      >
        <BrowserRouter>
          <UserLocationProvider>
            <ThemeProvider theme={context?.theme || MockTheme}>
              <OptionalVideoModalContextProvider>
                <ModelExplorationContextProvider>
                  <ModelFiltersContextProvider>{children}</ModelFiltersContextProvider>
                </ModelExplorationContextProvider>
              </OptionalVideoModalContextProvider>
            </ThemeProvider>
          </UserLocationProvider>
        </BrowserRouter>
      </SitecoreContext>
    );
  };
/**
 * A custom implementation of the testing-library `render` function that
 * wraps tests in an typical JSS app's context providers.
 * Note that you can include a `context` object to set or override values
 * for SitecoreContext and ThemeContext.
 * @param {ReactElement} component - the component you want testing-library to render
 * @param {object} [options] - see https://testing-library.com/docs/react-testing-library/api/#render-options
 * @param {string} [options.context] - default values for SitecoreContext, ThemeProvider
 * @param {string} [options.context.sitecoreContext] - value for SitecoreContext
 * @param {string} [options.context.theme] - value for ThemeProvider
 * @example
 * const { getByText } = jssRender(
 *  <MyComponent />, { context: { sitecoreContext: { language: 'fr' } } }
 * );
 * @returns {function(): void} custom testing-library render function
 */
const jssRender = (component, options = {}) => {
  const { context } = options;
  // eslint-disable-next-line no-param-reassign
  delete options.context;

  return render(component, {
    wrapper: generateJssWrapper(context),
    ...options,
  });
};

const generateWrapper = context =>
  // eslint-disable-next-line react/display-name
  function ({ children }) {
    return (
      <BrowserRouter>
        <ThemeProvider theme={context?.theme || MockTheme}>
          <OptionalVideoModalContextProvider>{children}</OptionalVideoModalContextProvider>
        </ThemeProvider>
      </BrowserRouter>
    );
  };

/**
 * A custom implementation of the testing-library `render` function that
 * wraps tests in typical design-system context providers.
 * Note that you can include a `theme` object to set or override values
 * for ThemeContext.
 * @param {ReactElement} component - the component you want testing-library to render
 * @param {object} [options] - see https://testing-library.com/docs/react-testing-library/api/#render-options
 * @param {string} [options.theme] - default value for ThemeProvider
 * @example
 * const { getByText } = render(
 *  <MyComponent />, { theme: { space: { wrapperGap: '15px' } } }
 * );
 * @returns {function(): void} custom testing-library render function
 */
const customRender = (component, options = {}) => {
  const { theme } = options;
  // eslint-disable-next-line no-param-reassign
  delete options.theme;

  return render(component, {
    wrapper: generateWrapper(theme),
    ...options,
  });
};

// disable accessibility checks for errors that are usually trigged by content mistakes
const axe = configureAxe({
  rules: {
    'duplicate-id': { enabled: false },
    'heading-order': { enabled: false },
  },
});

export * from 'jest-axe';
export { axe };
export * from '@testing-library/react';
export { customRender as render, jssRender };
export { MockFields, MockRendering, MockTheme };
