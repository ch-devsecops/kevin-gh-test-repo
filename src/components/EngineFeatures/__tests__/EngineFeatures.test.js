import React from 'react';
import { jssRender, cleanup } from 'test-utils';
import EngineFeatures from '../index';
import sitecoreContext from '../../../sitecoreContextMocks/sitecoreContext';
import mockTextFeatures from './__mocks__/textFeature.json';

afterEach(cleanup);

const mockData = require('../../../../sample-mock-data/routes/engine-features/en.json');
const mockEmptyData = require('./__mocks__/empty_en.json');

jest.mock('../../ProductFeaturesProvider', () => ({
  __esModule: true,
  A: true,
  default: ({ children }) => (children ? children({ features: mockTextFeatures }) : null),
}));

describe('<EngineFeatures>', () => {
  test('should render correctly', () => {
    if (!''.matchAll) {
      // this is for jenkins while running node < 12.0
      expect(true).toBe(true);
      return;
    }

    const { queryByText } = jssRender(<EngineFeatures fields={mockData} />, {
      context: { sitecoreContext },
    });

    const value = queryByText('Fuel-Efficient, High-Output Operation');

    expect(value).toBeInTheDocument();
  });

  test('should return null if there is no fields', () => {
    const { container } = jssRender(<EngineFeatures fields={null} />, { context: { sitecoreContext } });

    expect(container).toMatchSnapshot();
  });

  test('should render even if there are empty fields', () => {
    const { container } = jssRender(<EngineFeatures fields={mockEmptyData} />, {
      context: { sitecoreContext },
    });

    expect(container).toMatchSnapshot();
  });
});
