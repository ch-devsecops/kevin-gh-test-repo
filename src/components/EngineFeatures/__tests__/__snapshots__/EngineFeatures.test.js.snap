// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EngineFeatures> should render even if there are empty fields 1`] = `<div />`;

exports[`<EngineFeatures> should return null if there is no fields 1`] = `<div />`;
